scripts/check_copyright_header/src/main.rs

```diff
@@ -0,0 +1,150 @@
+use std::{
+    env,
+    fs::File,
+    io::{BufRead, BufReader},
+};
+
+fn check_file_header(lines: impl IntoIterator<Item = String>) -> Result<(), &'static str> {
```

Comment:
It is more common to use the crate `anyhow` for top-level programs: `anyhow::Result<()>`, etc